app/src/main/res/values/strings.xml

```diff
@@ -282,4 +282,51 @@
   <string name="notes_deletion_none_found">No notes found for deletion</string>
   <string name="notes_deletion_successful">Entire notes folder deleted</string>
   <string name="notes_deletion_unsuccessful">Some files not deleted</string>
-</resources>
\ No newline at end of file
+  <string name="discovery_initiated">Discovery Initiated</string>
+  <string name="discovery_failed">Discovery Failed:</string>
+  <string name="wifi_p2p_internal_error">Internal error</string>
+  <string name="wifi_p2p_framework_busy">Framework busy, unable to service request</string>
+  <string name="wifi_p2p_unsupported">P2P unsupported on this device</string>
+  <string name="wifi_p2p_unknown_error">Unknown error code -</string>
+  <string name="channel_lost">Channel lost, trying again</string>
+  <string name="severe_loss_error">Severe! Try Disable/Re-enable WiFi P2P</string>
+  <string name="abort_connection">Aborting connection</string>
+  <string name="abort_failed">Connect abort request failed. Reason:</string>
+  <string name="connection_failed">Connection failed:</string>
+  <string name="permission_rationale_location">Location permission is required by Android to allow the app to detect peer devices</string>
+  <string name="permission_rationale_storage">Storage permissions required for accessing and storing ZIM files</string>
+  <string name="permission_refused_location">Cannot locate peer devices without location permissions</string>
+  <string name="permission_refused_storage">Cannot access zim files without storage permission</string>
+  <string name="request_enable_location">Enable location to allow detection of peers</string>
+  <string name="open_location_settings">Open Location Settings</string>
+  <string name="request_refused_location">Cannot discover peers without location services</string>
+  <string name="request_enable_wifi">Enable WiFi P2P from system settings</string>
+  <string name="open_wifi_settings">Open WiFi Settings</string>
+  <string name="request_refused_wifi">Cannot discover peers without WiFi ON</string>
+  <string name="transfer_to">Transfer files to</string>
+  <string name="available">Available</string>
+  <string name="invited">Invited</string>
+  <string name="connected">Connected</string>
+  <string name="failed">Failed</string>
+  <string name="unavailable">Unavailable</string>
+  <string name="unknown">Unknown</string>
+  <string name="device_not_cooperating">Selected device not cooperating for transfer</string>
+  <string name="preparing_to_receive">Preparing to receive files...</string>
+  <string name="starting_transfer">Starting file transfer...</string>
+  <string name="discovery_needs_wifi">Cannot discover peers without WiFi</string>
+  <string name="file_transfer_complete">File transfer complete</string>
```

Comment:
This is a lot of extra strings for a "small" addition.

Reply:
It's a big module with a lot of callbacks going on, so the user needs to be informed whenever something changes
The strings here are for:
- Icon tool tips (given upon long pressing an entity on the screen)
- Asking for permissions
- Showing status of the device for WiFi Direct
- Giving updates about the connection process and then file-transfer

Reply:
I feel this is a lot of strings to translate for such a relative confined feature of the app. Is there any possibility of perhaps simplification the UI a bit so that they are not needed?

Reply:
I'll see where I can simplify and avoid using some :+1:

Reply:
@mhutti1 [this commit](https://github.com/kiwix/kiwix-android/pull/1276/commits/9a2bee7732cb749d9387e139f97268599cb8b689) is the most I could do to reduce strings required without doing a major UI rework. Other than a bunch of toasts ([here](https://github.com/kiwix/kiwix-android/pull/1276/commits/531b18481062c0fff76a1989d7efe9d88a2ca18c) and [here](https://github.com/kiwix/kiwix-android/pull/1276/commits/b4183a9b86f09f0c20ad642d2b45228d497c7214)) I have also removed ['device status'](https://github.com/kiwix/kiwix-android/pull/1276/commits/79e6078ef9bc5a7fc924091de87c3df68fbcc6a6) and related strings, and made UI changes to accommodate for the removals.
If more strings are to be removed, I'll have to then rethink the UI.